DNN Platform/Library/Entities/Modules/ModuleController.cs

```diff
@@ -250,6 +251,90 @@ private void CopyTabModuleSettingsInternal(ModuleInfo fromModule, ModuleInfo toM
             }
         }
 
+        /// <summary>
+        /// Checks whether module VIEW permission is inherited from its tab
+        /// </summary> 
+        /// <param name="module">The module</param>
+        /// <param name="permission">The module permission.</param>
+        private bool IsModuleViewPermissionInherited(ModuleInfo module, ModulePermissionInfo permission)
+        {
+            Requires.NotNull(module);
+
+            Requires.NotNull(permission);
+
+            var permissionViewKey = "VIEW";
+
+            if (!module.InheritViewPermissions || permission.PermissionKey != permissionViewKey)
+            {
+                return false;
+            }
+
+            var tabPermissions = TabPermissionController.GetTabPermissions(module.TabID, module.PortalID);
+            
+            return tabPermissions?.Where(x => x.RoleID == permission.RoleID && x.PermissionKey == permissionViewKey).Any() == true;
+        }
+
+        /// <summary>
+        /// Checks whether given permission is granted for translator role
+        /// </summary>
+        /// <param name="permission">The module permission.</param>
+        /// <param name="portalId">The portal ID.</param>
+        /// <param name="culture">The culture code.</param>
+        private bool IsTranslatorRolePermission(ModulePermissionInfo permission, int portalId, string culture)
+        {
+            Requires.NotNull(permission);
+
+            if (string.IsNullOrWhiteSpace(culture) || portalId == Null.NullInteger)
+            {
+                return false;
+            }
+
+            var translatorSettingKey = $"DefaultTranslatorRoles-{ culture }";
+
+            var translatorSettingValue = 
+                PortalController.GetPortalSetting(translatorSettingKey, portalId, null) ?? 
+                HostController.Instance.GetString(translatorSettingKey, null);
+
+            var translatorRoles =
+                translatorSettingValue?.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries) ?? Enumerable.Empty<string>();
+
+            return translatorRoles.Any(r => r.Equals(permission.RoleName, StringComparison.OrdinalIgnoreCase));

```

Comment:
You don't need to use "Enumerable.Empty()" below you can instead write all these like in the following

`return true == translatorRoles?.Any(r => r.Equals(permission.RoleName, StringComparison.OrdinalIgnoreCase));`

if it is null it will get resolved to `null == true` which will give the same result quickly. Or if you want you can do it in one-liner:

```
return true == translatorSettingValue?.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries)
    .Any(r => r.Equals(permission.RoleName, StringComparison.OrdinalIgnoreCase));
```

You already did the very same thing on line 274 below

Reply:
I agree. But both variants are workable. Why suggested solution is better?

Reply:
You are generating Enumerable.Empty<string>() just to work around for null value which is redundant